x-pack/plugins/lens/public/persistence/filter_references.ts

```diff
@@ -22,14 +22,18 @@ export function extractFilterReferences(
       type: 'index-pattern',
       id: filterRow.meta.index,
     });
-    return {
+    const newFilter = {
       ...filterRow,
       meta: {
         ...filterRow.meta,
         indexRefName: refName,
-        index: undefined,
       },
     };
+    // remove index because it's specified by indexRefName
+    delete newFilter.meta.index;
+    // remove value because it can't be persisted
+    delete newFilter.meta.value;
```

Comment:
An alternative way for it, which may reduce the number of lines in case we have to add more is using destructuring:
```ts
const { index, value, ...filteredMeta } = filterRow.meta;

return {
  ...filterRow,
  meta: {
    ...filteredMeta,
    indexRefName: refName
  }
};
```

Just a stylistic thing, so not a blocker for merge. 👍